src/Currency.php

```diff
@@ -25,7 +27,7 @@ final class Currency implements JsonSerializable
     /** @psalm-param non-empty-string $code */
     public function __construct(string $code)
     {
-        $this->code = $code;
+        $this->code = strtoupper($code);
```

Comment:
@Ocramius Are you willing to explain me why psalm is OK with this? Only if I add `assert(! empty($code))` is allows it. I would expect it to be allowed anyway since the non-emptiness is already checked during construction via the `non-empty-string` param, no?